### PR TITLE
Adding documentation on conventions for Analyzers.

### PR DESCRIPTION
**Pull request by jmarolf**
This adds the details about analyzer conventions in nuget packages

### CODE REVIEW

File: NuGet.Docs/Create/Analyzers-Conventions.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿##Analyzer Nuget Format##
+The proposed format should be identical to the format described [here](https://docs.nuget.org/Create/Enforced-Package-Conventions), except the specifiers in the path describe development host dependencies instead of build-time.
+
+$/analyzers/{**Framework Name**}{**Version**}/{**Supported Architecture**}/**{Supported Programming Language**}/{**Analyzer**}.dll
+
+**Framework Name and Version** the *optional* API surface area of the .NET framework that the contained dlls need to run. If no target is specified, dlls are assumed to apply to *all* targets
```
Comment (yishaigalatzer) [inline NuGet.Docs/Create/Analyzers-Conventions.md]: I understand how you can drop one of the items in the path, but not how you can drop multiple of them, and somehow figure out what was dropped.

Either we say that FrameworkName is All (for all frameworks), and similarly for versions/architecture
or we have a very complicated resolution rules.

Can you please elaborate?

**Comment by yishaigalatzer** [comment]
CC @hmemcpy

File: NuGet.Docs/Create/Analyzers-Conventions.md (hunk 1 above)
Comment (jmarolf) [inline NuGet.Docs/Create/Analyzers-Conventions.md]: If there is no folder, we assume that the current host can load the dll.  If the user places the dll in a folder whose framework is unsupported or unknown, we skip loading the dll.  If the framework is known and compatible, we load the dll.

Comment (yishaigalatzer) [inline NuGet.Docs/Create/Analyzers-Conventions.md]: If you just have cs, is that a language or a framework? There is no way to tell other than hardcoding cs

Comment (jmarolf) [inline NuGet.Docs/Create/Analyzers-Conventions.md]: I do not anticipate there being a "cs" framework.  I am ok special casing "cs", "vb", and "fs".  Under what circumstance would this happen?

Comment (yishaigalatzer) [inline NuGet.Docs/Create/Analyzers-Conventions.md]: OK, lets come up with exact samples of what collapsing scenarios are supported, and take it from there.

Other than that this looks really good.

**Comment by yishaigalatzer** [comment]
One pretty major thing I see that is missing, is how to build a package that works both for packages.config scenarios and project.json scenario. I think you made a mix of msbuild targets/props and install.ps1 to solve it. It is advisable to show the end users how to build it.

**Comment by jmarolf** [comment]
ping @yishaigalatzer

**Comment by yishaigalatzer** [comment]
Thanks for the update and sorry I missed the fact that you pushed a new one. Added some comments, overall looks close

**Comment by yishaigalatzer** [comment]
@jmarolf ping

**Comment by jmarolf** [comment]
@yishaigalatzer Is there anything else you want added that was not in the last two commits?  Anything worded improperly?

**Pull request merged**